### DISCUSSION TITLE
Why prisma does have custom language for schema

### DISCUSSION THREAD

**Discussion by ryskajakub** [prisma/prisma#16660] [Q&A] [answered]
Hi all, 

does anyone know, what was the design decision, why there is this custom language for defining schema? Why not just define the entities in the code. I would maybe get it if it wouldn't be just nodejs project, then people wouldn't need to learn typescript, but it's not the case ...

**Comment by nurul3101** [comment, marked as answer]
Hey @ryskajakub 👋 

Prisma Schema is the main configuration file for your Prisma setup, it is utilised not only for Node.js runtime, but there are other community-maintained clients like [Prisma Client Rust](https://github.com/Brendonovich/prisma-client-rust) and [Prisma Client Python](https://prisma-client-py.readthedocs.io/en/stable/) which utilizes this schema file, and in future, there is a possibility for supporting other clients as well so creating a single source of truth in form of a schema file for all these clients makes sense.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
